ddtrace/contrib/asgi/middleware.py

```diff
@@ -70,13 +70,12 @@ def _default_handle_exception_span(exc, span):
 def span_from_scope(scope):
     # type: (Mapping[str, Any]) -> Optional[Span]
     """Retrieve the top-level ASGI span from the scope."""
-    return scope.get("datadog", {}).get("request_span")
+    return scope.get("datadog", {}).get("request_spans", [None])[0]
```

Comment:
@brettlangdon  sorry, I merged you're suggestion and the comment went away... how are we mixing types here / how can I better understand how that's a bad practice? Lastly, what do you think would be the right way to write this?